schema.development.kf

```diff
@@ -451,14 +448,36 @@ action get_credential_owned ($id) public view {
 @kgw(authn='true')
 procedure get_credential_shared ($id uuid) public view returns table (id uuid, human_id uuid, credential_type text,
     credential_level text, credential_status text, content text, encryption_public_key text, issuer text, inserter text) {
+    if !shared_credential_exist($id) {
+        error('shared credential does not exist');
+    }
+
+    if !data_granted_to_grantee($id, @caller) {
+        error('shared credential is not shared with the caller');
+    }
+
     if !has_grants($id) {
-        error('caller does not have access');
+        error('the credential has grant but the owner does not match');
```

Comment:
How should we handle when the owner has deleted their original credential, but the grantee still has a valid copy/AG?

Reply:
Sorry i was confused by the code line commented. Your question is valid but for [this line](https://github.com/trustfractal/idos-schema/blob/c7480bc10591cb6471f4470786f1597b8a49181c/schema.development.kf#L474)
Current implementation of the `shared_credential_exist` will fail because of deleting original credential also deletes `shared_credentials` record. The intention of doing such check was to catch the situation when a forged AG has `data_id` of **original** credential but not a duplicate.
I am not sure do we need to such check or better remove it in favor of proper handling case when original credential was deleted.

Reply:
We talked over voice. We decided to add something extra for an AG to survive the deletion of the original credential.